src/source/Ice/IceAgentStateMachine.c

```diff
@@ -222,7 +222,6 @@ STATUS fromGatheringIceAgentState(UINT64 customData, PUINT64 pState)
     // move to failed state if any error happened.
     CHK_STATUS(pIceAgent->iceAgentStatus);
 
-    CHK_STATUS(iceAgentStateMachineCheckDisconnection(pIceAgent, &state));
```

Comment:
Why remove all the `iceAgentStateMachineCheckDisconnection` calls? I am not familiar with the code really sorry

Reply:
the issue is that the disconnection detection is coinciding with the state transition timeouts. For example, if no remote candidate is received, we would report disconnection instead of failure in say ice check connection state. I didnt remove all disconnection check though. Now it's only checked in ice ready state. So we only check for disconnection after we have a selected pair.